Create Confirmation Popup

alaasaedy opened the pull request:
- Create Popup component using reactjs-popup package.

alaasaedy committed:
Creating Confirmation Popup

alaasaedy committed:
adding more styles to align with website design

MAlngaawy reviewed (changes requested):
nice work Alaa :+1: 
please, consider these requirements.

**styles**
- handle the responsive screens, don't make it fill the width 
- fix the padding issue in the mobile screens
- Cancel button should be a gray background and bold black text.
- the confirm button shouldn't be yes, it should be descriptive, like (delete, report) and his text and background color should come as props to it.
red in the delete and black in the report
- confirm button should take the incoming function, prop or delete when click should execute the function and then close the popup

**props should be in this component**
- the confirm text
- the confirm button text (delete, report, etc...)
- the confirm button color
- the confirm function

**and finally, implement it with the delete and report options**

alaasaedy committed:
Merge remote-tracking branch 'origin' into component/confirmation-popup

alaasaedy committed:
- Add margin and padding to box on mobile screen.
- Change cancel font weight and background.
- Add required props.
- Add function to close the popup and log "Deleted", when clicking on confirm.

alaasaedy committed:
- Fix svg's wrong attr (fillRule, clipRule)
- Add Popup component to Options button to report or delete post

MAlngaawy committed:
fix some popup issues

MAlngaawy committed:
remove unused code

alaasaedy committed:
Fixing logic issue

MAlngaawy committed:
add posts localhost instead of live link

MAlngaawy committed:
Merge branch 'develop' of https://github.com/m3ntorship/pickly-frontend into component/confirmation-popup

MAlngaawy reviewed (approved):
LGTM
